cmd/server.go

```diff
@@ -99,17 +119,24 @@ func (s *Server) securityCredentialsHandler(w http.ResponseWriter, r *http.Reque
 
 func (s *Server) roleHandler(w http.ResponseWriter, r *http.Request) {
 	remoteIP := parseRemoteAddr(r.RemoteAddr)
-	allowedRole, err := s.getRole(remoteIP)
-	allowedRoleARN := s.iam.roleARN(allowedRole)
+	podRole, err := s.getRole(remoteIP)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusNotFound)
 		return
 	}
+	podRoleARN := s.iam.roleARN(podRole)
+
+	if !s.store.CheckNamespaceRestriction(podRoleARN, remoteIP) {
+		http.Error(w, fmt.Sprintf("Role requested %s not valid for namespace of pod at %s", podRole, remoteIP), http.StatusNotFound)
```

Comment:
We don't actually store/track the pod name anywhere. Perhaps we could do this in another PR.